internal/controllers/inventory_controller.go

```diff
@@ -896,7 +896,7 @@ func (t *reconcilerTask) createService(ctx context.Context, resourceName string)
 	}
 
 	t.logger.InfoContext(ctx, "[createService] Create/Update/Patch Service: ", "name", resourceName)
-	if err := utils.CreateK8sCR(ctx, t.client, newService, t.object, utils.PATCH); err != nil {
+	if err := utils.CreateK8sCR(ctx, t.client, newService, t.object, utils.UPDATE); err != nil {
```

Comment:
Off topic/general comment...`CreateK8sCR` seems to do way too many things right? One func to do create, update, patch, taking ownership, doing all sorts of reflection...

Reply:
The purpose was to create and update objects in the same function so that the main reconciliation loop is cleaner. When you create an object, it's when you also specify its ownership, so these should be together anyway.
The inventory controller makes sure the servers exist at all times (so it recreates them if they've been deleted) and that they look as specified in the Inventory CR spec (so it updates/patches if someone edits them directly).
I suppose we could separate the updates from creation in different functions.

What enhancement did you have in mind for this?

Reply:
Few suggestions: 
- We cant say "Create" and then take an operation parameter (which happens to an arbitrary string) 
- This is not simply "Create", violating single responsibility principle. Instead we break it down into `updateOrPatchObject` `createObject` etc etc. And then finally the caller should explicitly check for errors as needed e.g. `k8srr.IsAlreadyExists(err)` for create
- Given we already have access all the generated for all CRUD reqs....we should try to use them directly instead of figure out types at runtime with `Object` . i.e c.Create(ctx, job) (job being a struct type job). This means more code but idiomatic/in-line with Go

Reply:
What's going to happen if we split it into a `updateOrPatchObject` and a `createObject` is that someone will get tired of seeing that pattern spread throughout the code and will create a write for that flow called `createOrUpdateObject` and we're more or less back to this starting point.   

I do agree that the name could be approved upon, and certain how we use it will need to be addressed in the future because as of now the 'update' is rather unsophisticated and doesn't handle needing to modify objects very gracefully at all.   They all end up being just a bulk replace of the object which is likely causing issues to anyone doing a watch on them (potentially)

Reply:
I agree that renaming it to smth like `ensureK8sCR` would be a better description of the functionality.
I don't really agree that this is violating the SRP principle, especially with the renaming and possibly enhancing the update.

Reply:
Yeah definitely doing `createObject` would ultimate be wrapped up. Which is why it's better to be explicit (the third point) and pass the exact typed...so instead we should have `createJob` or `createDeployment` instead of super generic funcs